### DISCUSSION TITLE
How to add sentence boundaries in Json

### DISCUSSION THREAD

**Discussion by grivaz** [explosion/spaCy#8345] [Help: Coding & Implementations] [answered]
I am generating my own json corpus for training, and then using spacy convert to convert it into importable spacy format e.g.
`python -m spacy convert test.json test`

I cannot seem to properly add sentence boundaries. I tried all variation of setting sent_start or is_sent_start on the tokens, but it doesn't seem to work. Here is the beginning o fan example json with setting both.

> [
>     {
>         "id": 0,
>         "paragraphs": [
>             {
>                 "sentences": [
>                     {
>                         "tokens": [
>                             {
>                                 "id": 1,
>                                 "dep": "dummy",
>                                 "head": 1,
>                                 "tag": "DT",
>                                 "orth": "The",
>                                 "sent_start": true,
>                                 "is_sent_start": true
>                             },
>                             {
>                                 "id": 2,
>                                 "dep": "A1",
>                                 "head": 2,
>                                 "tag": "NN",
>                                 "orth": "economy",
>                                 "sent_start": false,
>                                 "is_sent_start": false
>                             },

I then load the converted .spacy file and look at the tokens in a debugger, and see that for the first token is_sent_start is False and sent_start is False. For the second token is_sent_start is False and sent_start is -1. How do I properly set sentence boundaries in json?

**Comment by adrianeboyd** [comment, marked as answer]
Hi, the `sentences` key should be a list of dicts, where each dict corresponds to one sentence, so you would have:

```none
"sentences": [
{"tokens": [...]},
{"tokens": [...]},
{"tokens": [...]}
]
```

You can also use `-t json` with `spacy convert` to convert to output JSON instead of DocBin.

**Reply by grivaz** [reply]
Thank you so much for your quick answer!
